map.go

```diff
@@ -515,7 +508,8 @@ func newMapIterator(target *Map) *MapIterator {
 //
 // Returns false if there are no more entries.
 //
-// See Map.Get for further caveats around valueOut.
+// KeyOut mustn't be an unsafe.Pointer. See Map.Get for further
```

Comment:
This is a bit of a wart. Allowing unsafe.Pointer as the key is kind of difficult, since the iterator needs to retain a copy of the last key as a byte slice. This means that we have to copy from that slice to the unsafe.Pointer, which isn't entirely trivial.

I'm not sure whether it's even useful to support `unsafe.Pointer` in this case at all, so I chose to error out and fix this once somebody complains.

Reply:
I've fixed this, it turns out to not be much code. More unsafe :(